### PR TITLE
new HiGHS interface

### PR DESCRIPTION
**Pull request by Transurgeon**
## Description
Please include a short summary of the change.
The great open source LP/QP/MILP solver HiGHS has its own python interface ``highspy`` which allows it to be easily interfaced through cvxpy. This PR aims to add a new conic interface to HiGHS (where it was previously being called indirectly through the SciPy solver). This PR also explicitly adds HiGHS to the cvxpy documentation so that users can be aware of it. 
Issue link (if applicable): #2568 

## Type of change
- [x] New feature (backwards compatible)
- [ ] New feature (breaking API changes)
- [ ] Bug fix
- [ ] Other (Documentation, CI, ...)

## [Contribution checklist](https://www.cvxpy.org/contributing/index.html#contribution-checklist)
- [x] Add our license to new files.
- [x] Check that your code adheres to our coding style.
- [x] Write unittests.
- [x] Run the unittests and check that they’re passing.
- [x] Run the benchmarks to make sure your change doesn’t introduce a regression.

### CODE REVIEW

**Comment by PTNobel** [comment]
I think you should update this as well: https://github.com/cvxpy/cvxpy/blob/8eb302c8b9b365a755cf4db0ccd274fd0da4c33a/pyproject.toml#L73

File: doc/source/tutorial/solvers/index.rst [hunk 1]
```diff
@@ -91,6 +91,8 @@ The table below shows the types of problems the supported solvers can handle.
 +----------------+----+----+------+-----+-----+-----+-----+
 | `GUROBI`_      | X  | X  | X    |     |     |     | X   |
 +----------------+----+----+------+-----+-----+-----+-----+
+| `HiGHS`_       | X  | X  |      |     |     |     | X*  |
```
Comment (SteveDiamond) [inline doc/source/tutorial/solvers/index.rst:94]: does HIGHS really solve QPs? That's not my understanding.

Comment (Transurgeon) [reply doc/source/tutorial/solvers/index.rst:94]: I saw this in the docs, but not sure since I have never used it before.
image

Comment (SteveDiamond) [reply doc/source/tutorial/solvers/index.rst:94]: I see. The interface right now only accepts LPs though. I would be fine with just supporting MILPs to start. We can add in QP support later. Or if you have the time you could switch the interface to be a QP solver interface.

Comment (Transurgeon) [reply doc/source/tutorial/solvers/index.rst:94]: ok sounds good, let me update the docs

**Comment by Transurgeon** [comment]
closing as another PR has taken over.

**Pull request closed without merging**